Public/Src/Tools/DropDaemon/DropDaemon.cs

```diff
@@ -567,43 +569,73 @@ private async Task ReportStatisticsAsync()
             }
         }
 
+        private delegate TResult ErrorFactory<TResult>(string message, IpcResultStatus status);
+
         private static Task<IIpcResult> WrapDropErrorsIntoIpcResult(Func<Task<IIpcResult>> factory)
         {
-            return HandleKnownErrors(
+            return HandleKnownErrorsAsync(
                 factory,
-                (errorMessage) => new IpcResult(IpcResultStatus.ExecutionError, errorMessage));
+                (errorMessage, status) => new IpcResult(status, errorMessage));
         }
 
         private static Task<TDropEvent> WrapDropErrorsIntoDropEtwEvent<TDropEvent>(Func<Task<TDropEvent>> factory) where TDropEvent : DropOperationBaseEvent
         {
-            return HandleKnownErrors(
+            return HandleKnownErrorsAsync(
                 factory,
-                (errorMessage) =>
+                (errorMessage, errorKind) =>
                 {
                     var dropEvent = Activator.CreateInstance<TDropEvent>();
                     dropEvent.Succeeded = false;
                     dropEvent.ErrorMessage = errorMessage;
+                    dropEvent.AdditionalInformation = RenderIpcStatus(errorKind);
                     return dropEvent;
                 });
         }
 
-        private static async Task<TResult> HandleKnownErrors<TResult>(Func<Task<TResult>> factory, Func<string, TResult> errorValueFactory)
+        private static string RenderIpcStatus(IpcResultStatus status)
+        {
+            return status.ToString();
+        }
+
+        private static IpcResultStatus ParseIpcStatus(string statusString, IpcResultStatus defaultValue = IpcResultStatus.ExecutionError)
+        {
+            return Enum
+                .GetValues(typeof(IpcResultStatus))
+                .Cast<IpcResultStatus?>()
+                .Where(status => status.ToString() == statusString)
+                .FirstOrDefault() ?? defaultValue;

```

Comment:
>?? defaultValue [](start = 34, length = 15)

Should we add a new IpcResultStatus value that would be used specifically for parse failures? I'm afraid that we might be 'hiding' parsing problems here, no?

Reply:
I changed it to use `TryParse`.  I am pretty confident that `ToString`/`TryParse` are the exact inverses of each other in this case, so I'd leave it as is